File: 30_days_of_code/8-dictionaries-and-maps/py/dictionaries-and-maps.py
#!/bin/python

import sys

n = int(raw_input().strip())
phone_book = {}

for x in range(0, n):
    [name, phone] = raw_input().strip().split(' ')
    phone_book[name] = phone

while True:
    try:
        input = raw_input().strip()
    except EOFError:
        break

    if input in phone_book:
        sys.stdout.write(input + "=" + phone_book[input] + "\n")
    else:
        print "Not found"

File: 30_days_of_code/8-dictionaries-and-maps/elixir/dictionaries-and-maps.ex
defmodule Solution do
  def loop_search(:eof, _) do
  end

  def loop_search(value, book) do
    value = String.rstrip(value)
    if book[value] do
      IO.puts "#{value}=#{book[value]}"
    else
      IO.puts "Not found"
    end
    loop_search(IO.read(:stdio, :line), book)
  end
end

{n, _} = Integer.parse(IO.gets(""))

phone_book = Enum.reduce((1..n), %{}, fn(_, acc) ->
  [name, phone] = String.split(String.strip(IO.gets("")))
  Map.put(acc, name, phone)
end)

Solution.loop_search(IO.read(:stdio, :line), phone_book)
